### ISSUE TITLE
Can't find node-cmake

### ISSUE THREAD

**Issue by mitch-besser** [mapbox/mapbox-gl-native#6847]
I'm attempting to build the source using Android Studio and CMake, but I'm getting the error:

"Can't find node-cmake"

I see the code in: https://github.com/mapbox/mapbox-gl-native/blob/master/cmake/mbgl.cmake

if(NOT EXISTS ${CMAKE_SOURCE_DIR}/node_modules/node-cmake/FindNodeJS.cmake)
    message(FATAL_ERROR "Can't find node-cmake")
endif()

I don't have the FindNodeJS.cmake file. Does anyone know how I can find it? Am I using the wrong branch?

Thanks.

**Comment by jfirebaugh** [comment]
It should have been installed by the build scripts. Are you following the Android build instructions in [CONTRIBUTING_LINUX.md](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/CONTRIBUTING_LINUX.md) or [CONTRIBUTING_MACOS.md](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/CONTRIBUTING_MACOS.md) for your platform?

**Comment by mitch-besser** [comment]
I got a new install version from a CMake expert (by expert, I mean someone that knows far more than myself). I didn't see the file, but I also looked here on Github for the file and didn't see it. I can try the standard install and see if it shows up. If it does, then I'll go back to my "expert" and ask why that file is missing. I thought maybe I had the wrong branch. I'm guessing you're saying I should have gotten it on the Master branch?

**Comment by mitch-besser** [comment]
I just installed again on a different computer. I followed the clone instruction as described, but I don't see the FindNodeJS.cmake downloaded.

**Comment by jfirebaugh** [comment]
Can you please post the full log of the commands you executed and their output?

**Comment by mitch-besser** [comment]
Not sure what you mean by the "full log". The first command I execute is:

git clone https://github.com/mapbox/mapbox-gl-native.git

Then I searched for the FindNodeJS.cmake file, but the recursive search doesn't find a file by this name. Should I expect to find it?  

I can show you the CMake output when I get home this evening. I don't think it will tell you much except that it creates the error I described above in that it can't find the FindNodeJS.cmake file. 

Regardless, when I get home I can post the output from build. 

Meanwhile, can you confirm master is the correct branch and that I should have a copy of FindNodeJS.cmake after I clone from Githhub? It could be as simple as the wrong branch for working with CMake.

**Comment by jfirebaugh** [comment]
> Then I searched for the FindNodeJS.cmake file, but the recursive search doesn't find a file by this name. Should I expect to find it?

Not immediately after cloning. It's created later in the build process. That's why I ask for the exact commands you're running and their output.

**Comment by kkaefer** [comment]
@mitch-besser you'll have to run one of the Make targets, in your case `make android`. This will ensure that all of the dependencies are in place.

**Comment by bessermt** [comment]
hi @kkaefer , 

I'm not using make, I'm using CMake. I'll post the CMakelists.txt in the next post for clarity, Here's the error:

Error:FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:externalNativeBuildDebug'.
> Build command failed.
  Error while executing '/home/bessermt/Android/Sdk/cmake/3.6.3155560/bin/cmake' with arguments {--build /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64 --target native-lib}
  [1/1] Re-running CMake...
  -- Configuring done
  -- Generating done
  -- Build files have been written to: /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64
  [1/11] Creating directories for 'mapbox-gl'
  [2/11] Performing download step (git clone) for 'mapbox-gl'
  -- Avoiding repeated git clone, stamp file is up to date: '/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl-stamp/mapbox-gl-gitclone-lastrun.txt'
  [3/11] Performing update step for 'mapbox-gl'
  [4/11] No patch step for 'mapbox-gl'
  From github.com:mapbox/mapbox-gl-native
     eb1b385..62b56b7  master     -> origin/master
   * [new branch]      1ec5-string-enum-6791 -> origin/1ec5-string-enum-6791
   * [new branch]      1ec5-換行-1223 -> origin/1ec5-換行-1223
   * [new branch]      5910-move-listener-logic-for-nougat -> origin/5910-move-listener-logic-for-nougat
   * [new branch]      6765-document-loc-padding-units -> origin/6765-document-loc-padding-units
   * [new branch]      6811-updates-directions-example -> origin/6811-updates-directions-example
   * [new branch]      6836-atlas-textures -> origin/6836-atlas-textures
   * [new branch]      fb-language-toggle -> origin/fb-language-toggle
   * [new branch]      fix-symbol-clipping -> origin/fix-symbol-clipping
   * [new branch]      fuzzy-satellite -> origin/fuzzy-satellite
   + ce34d4e...aecc40d gl-bindings -> origin/gl-bindings  (forced update)
   * [new branch]      ios-network-config-docs -> origin/ios-network-config-docs
   * [new branch]      query-feature-order -> origin/query-feature-order
   * [new branch]      release-android-v4.2.0 -> origin/release-android-v4.2.0
   * [new branch]      release-ios-v3.3.5 -> origin/release-ios-v3.3.5
     1216050..d5af9b6  release-ios-v3.4.0 -> origin/release-ios-v3.4.0
   * [new branch]      terrain-rendering -> origin/terrain-rendering
   * [new branch]      tobrun-instrumentation-tests -> origin/tobrun-instrumentation-tests
   + 6f3a220...acbb073 transformstate-helpers -> origin/transformstate-helpers  (forced update)
   * [new tag]         android-v4.2.0-beta.4 -> android-v4.2.0-beta.4
  First, rewinding head to replay your work on top of it...
  Fast-forwarded master to origin/master.
  [5/11] Performing configure step for 'mapbox-gl'
  FAILED: cd /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl-build && /home/bessermt/Android/Sdk/cmake/3.6.3155560/bin/cmake "-GAndroid Gradle - Ninja" /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl && /home/bessermt/Android/Sdk/cmake/3.6.3155560/bin/cmake -E touch /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl-stamp/mapbox-gl-configure
  CMake Error at cmake/mbgl.cmake:16 (message):
    Can't find node-cmake
  Call Stack (most recent call first):
    CMakeLists.txt:5 (include)
  
  
  -- Configuring incomplete, errors occurred!
  See also "/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/CMakeFiles/CMakeOutput.log".
  ninja: build stopped: subcommand failed.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

**Comment by bessermt** [comment]
# CMakeLists.txt
# Sets the minimum version of CMake required to build the native
# library. You should either keep the default value or only pass a
# value of 3.4.0 or lower.

cmake_minimum_required(VERSION 3.4.1)

include(ExternalProject)

ExternalProject_Add(mapbox-gl
                    GIT_REPOSITORY     git@github.com:mapbox/mapbox-gl-native.git
                    GIT_TAG            master
                    TEST_AFTER_INSTALL 1
                    TEST_COMMAND       killTheBuildHere
)

#ExternalProject_Add(mapbox-gl
#                   GIT_REPOSITORY git@github.com:mapbox/mapbox-gl-native.git
#                   GIT_TAG        master
#)

# Creates and names a library, sets it as either STATIC
# or SHARED, and provides the relative paths to its source code.
# You can define multiple libraries, and CMake builds it for you.
# Gradle automatically packages shared libraries with your APK.

add_library( # Sets the name of the library.
             native-lib

             # Sets the library as a shared library.
             SHARED

             # Provides a relative path to your source file(s).
             # Associated headers in the same location as their source
             # file are automatically included.
             src/main/cpp/native-lib.cpp )

# Searches for a specified prebuilt library and stores the path as a
# variable. Because system libraries are included in the search path by
# default, you only need to specify the name of the public NDK library
# you want to add. CMake verifies that the library exists before
# completing its build.

find_library( # Sets the name of the path variable.
              log-lib

              # Specifies the name of the NDK library that
              # you want CMake to locate.
              log )

add_dependencies(native-lib mapbox-gl)

# Specifies libraries CMake should link to your target library. You
# can link multiple libraries, such as libraries you define in the
# build script, prebuilt third-party libraries, or system libraries.

target_link_libraries( # Specifies the target library.
                       native-lib

                       # Links the target library to the log library
                       # included in the NDK.
                       ${log-lib} )

**Comment by bessermt** [comment]
Hi @jfirebaugh , 

Okay, got it. Don't expect the clone to download the missing FindNodeJS.cmake. Expect the build to do download or possibly create FindNodeJS.cmake. I assume the CMakeLists.txt should be doing this as posted above, but I don't know enough about CMake to know.  I can see the CMakeLists.txt has created some directories, but I'm not sure if it is creating everything it should. 

For example the documentation https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/CONTRIBUTING_LINUX.md says to run 

>cd mapbox-gl-native
>BUILDTYPE=Debug make android

except I don't see a mapbox-gl-native directory and I would hope the CMakeLists.txt would have created that folder during the download. True? If not, then in which folder would I execute the "BUILDTYPE=Debug make android" command?

**Comment by bessermt** [comment]
Not sure if this is helpful or not, but I made an attempt to run the "make android" command then build and got:

/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/include/mbgl/gl/gl.hpp
Error:(27, 23) fatal error: GL/gl.h: No such file or directory

Next I'll try the similar:

MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl $ BUILDTYPE=Debug make android

and see if that helps.

Totally guessing on the folder to run the command from.

**Comment by bessermt** [comment]
Nope. 

/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/include/mbgl/gl/gl.hpp
Error:(27, 23) fatal error: GL/gl.h: No such file or directory
Error:(27, 23) fatal error: GL/gl.h: No such file or directory
Error:(27, 23) fatal error: GL/gl.h: No such file or directory

**Comment by bessermt** [comment]
I've tracked it down to __ANDROID__ not being defined when I assume it should be? Could that be the issue?

**Comment by bessermt** [comment]
Hi @jfirebaugh and @kkaefer 

Shouldn't __ANDROID__ be defined to some non-zero value? If so, why would the compile be looking for the #include <GL/gl.h> in the #else portion? How should my build have defined __ANDROID__ to a non-zero value?

#elif __ANDROID__ || MBGL_USE_GLES2
    #define GL_GLEXT_PROTOTYPES
    #include <GLES2/gl2.h>
    #include <GLES2/gl2ext.h>
#else
    #define GL_GLEXT_PROTOTYPES
    #include <GL/gl.h>
    #include <GL/glext.h>
#endif

**Comment by jfirebaugh** [comment]
@bessermt What we need to help you effectively is what we have already asked for: the full logs of what you have tried, including the  *exact* commands you are running, starting from cloning the repository, and their output.

**Comment by bessermt** [comment]
Hi @jfirebaugh 

As I said in my response last time you asked: "Not sure what you mean by the "full log."

If you mean the CMakeLists.txt, that was already provided above. If you mean the output from compiling, that is also already provided above. If you mean how to I run the CMakeLists.txt, I use Android Studio using Android Studio instructions. [](https://developer.android.com/studio/projects/add-native-code.html)

My best guesses are I need to somehow 

define `__ANDROID__` 

and/or run the

`BUILDTYPE=Debug make android` 

command from a special in a CMake download folder. I don't see a generated mapbox-gl-native folder being generated by CMake during the Android Studio build, so I can't do that. 

Please explain what you mean by the "Full Log" and I will get that to you. Maybe you want the Android Studio console output?  I'll post that below in case that's helpful. The command I gave to produce this was 'Build | Make Project'.

Should I work on defining the `__ANDROID__` preprocessor symbol?  If so, how do I go about doing that?

```
Executing tasks: [:app:generateDebugSources, :app:generateDebugAndroidTestSources, :app:mockableAndroidJar, :app:prepareDebugUnitTestDependencies, :app:compileDebugSources, :app:compileDebugAndroidTestSources, :app:compileDebugUnitTestSources]

Configuration on demand is an incubating feature.
Incremental java compilation is an incubating feature.
:app:preBuild UP-TO-DATE
:app:preDebugBuild UP-TO-DATE
:app:checkDebugManifest
:app:preReleaseBuild UP-TO-DATE
:app:prepareComAndroidSupportAnimatedVectorDrawable2421Library UP-TO-DATE
:app:prepareComAndroidSupportAppcompatV72421Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCompat2421Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCoreUi2421Library UP-TO-DATE
:app:prepareComAndroidSupportSupportCoreUtils2421Library UP-TO-DATE
:app:prepareComAndroidSupportSupportFragment2421Library UP-TO-DATE
:app:prepareComAndroidSupportSupportMediaCompat2421Library UP-TO-DATE
:app:prepareComAndroidSupportSupportV42421Library UP-TO-DATE
:app:prepareComAndroidSupportSupportVectorDrawable2421Library UP-TO-DATE
:app:prepareDebugDependencies
:app:compileDebugAidl UP-TO-DATE
:app:compileDebugRenderscript UP-TO-DATE
:app:generateDebugBuildConfig UP-TO-DATE
:app:generateDebugResValues UP-TO-DATE
:app:generateDebugResources UP-TO-DATE
:app:mergeDebugResources UP-TO-DATE
:app:processDebugManifest UP-TO-DATE
:app:processDebugResources UP-TO-DATE
:app:generateDebugSources UP-TO-DATE
:app:preDebugAndroidTestBuild UP-TO-DATE
:app:prepareComAndroidSupportTestEspressoEspressoCore222Library UP-TO-DATE
:app:prepareComAndroidSupportTestEspressoEspressoIdlingResource222Library UP-TO-DATE
:app:prepareComAndroidSupportTestExposedInstrumentationApiPublish05Library UP-TO-DATE
:app:prepareComAndroidSupportTestRules05Library UP-TO-DATE
:app:prepareComAndroidSupportTestRunner05Library UP-TO-DATE
:app:prepareDebugAndroidTestDependencies
:app:compileDebugAndroidTestAidl UP-TO-DATE
:app:processDebugAndroidTestManifest UP-TO-DATE
:app:compileDebugAndroidTestRenderscript UP-TO-DATE
:app:generateDebugAndroidTestBuildConfig UP-TO-DATE
:app:generateDebugAndroidTestResValues UP-TO-DATE
:app:generateDebugAndroidTestResources UP-TO-DATE
:app:mergeDebugAndroidTestResources UP-TO-DATE
:app:processDebugAndroidTestResources UP-TO-DATE
:app:generateDebugAndroidTestSources UP-TO-DATE
:app:mockableAndroidJar UP-TO-DATE
:app:preDebugUnitTestBuild UP-TO-DATE
:app:prepareDebugUnitTestDependencies
:app:incrementalDebugJavaCompilationSafeguard UP-TO-DATE
:app:compileDebugJavaWithJavac UP-TO-DATE
:app:generateJsonModelDebug UP-TO-DATE
:app:externalNativeBuildDebug
  building /home/bessermt/dev/MyFirstMapboxApp/app/build/intermediates/cmake/debug/obj/mips64/libnative-lib.so
:app:externalNativeBuildDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:externalNativeBuildDebug'.
> Build command failed.
  Error while executing '/home/bessermt/Android/Sdk/cmake/3.6.3155560/bin/cmake' with arguments {--build /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64 --target native-lib}
  [1/8] Performing update step for 'mapbox-gl'
  From github.com:mapbox/mapbox-gl-native
     a4c82b8..39d5cb3  master     -> origin/master
   * [new branch]      shaders-refactor -> origin/shaders-refactor
  First, rewinding head to replay your work on top of it...
  Fast-forwarded master to origin/master.
  [2/8] Performing configure step for 'mapbox-gl'
  -- Configuring done
  -- Generating done
  -- Build files have been written to: /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl-build
  [3/8] Performing build step for 'mapbox-gl'
  FAILED: cd /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl-build && /home/bessermt/Android/Sdk/cmake/3.6.3155560/bin/cmake --build . && /home/bessermt/Android/Sdk/cmake/3.6.3155560/bin/cmake -E touch /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl-stamp/mapbox-gl-build
  [1/290] Building CXX object CMakeFiles/mbgl-loop.dir/platform/default/async_task.cpp.o
  [2/290] Building CXX object CMakeFiles/mbgl-loop.dir/platform/default/run_loop.cpp.o
  [3/290] Building CXX object CMakeFiles/mbgl-loop.dir/platform/default/timer.cpp.o
  [4/290] Generating generated/include/mbgl/util/version.hpp
  Tag: 0.0.0
  Rev: unknown
  [5/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/shader/symbol_uniforms.cpp.o
  [6/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/actor/mailbox.cpp.o
  [7/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/algorithm/generate_clip_ids.cpp.o
  [8/290] Linking CXX static library libmbgl-loop.a
  [9/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/annotation/annotation_manager.cpp.o
  [10/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/annotation/annotation_source.cpp.o
  [11/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/annotation/annotation_tile.cpp.o
  [12/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/annotation/fill_annotation_impl.cpp.o
  [13/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/annotation/line_annotation_impl.cpp.o
  [14/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/annotation/shape_annotation_impl.cpp.o
  [15/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/annotation/style_sourced_annotation_impl.cpp.o
  [16/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/annotation/symbol_annotation_impl.cpp.o
  [17/290] Building CXX object CMakeFiles/mbgl-core.dir/src/clipper/clipper.cpp.o
  [18/290] Building CXX object CMakeFiles/mbgl-core.dir/src/csscolorparser/csscolorparser.cpp.o
  [19/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/geometry/feature_index.cpp.o
  [20/290] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/geometry/line_atlas.cpp.o
  FAILED: /usr/bin/c++    -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/src -Igenerated/./include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/geometry/0.8.0/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/variant/1.1.0/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/unique_resource/dev/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/rapidjson/1.1.0/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/boost/1.60.0/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/geojson/0.3.2/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/geojsonvt/6.1.3/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/supercluster/0.2.0/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/kdbush/0.1.1/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/earcut/0.12.1/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/headers/protozero/1.4.2/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/platform/default -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/linux-x86_64/sqlite/3.14.2/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/linux-x86_64/nunicode/1.7.1/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/linux-x86_64/libpng/1.6.25/include/libpng16 -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/linux-x86_64/libjpeg-turbo/1.5.0/include -I/home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/mason_packages/linux-x86_64/webp/0.5.1/include -std=c++14 -ftemplate-depth=1024 -Wall -Wextra -Wshadow -Werror -Wno-variadic-macros -Wno-unknown-pragmas -fext-numeric-literals -g   -fPIC -fvisibility-inlines-hidden -std=gnu++14 -MD -MT CMakeFiles/mbgl-core.dir/src/mbgl/geometry/line_atlas.cpp.o -MF CMakeFiles/mbgl-core.dir/src/mbgl/geometry/line_atlas.cpp.o.d -o CMakeFiles/mbgl-core.dir/src/mbgl/geometry/line_atlas.cpp.o -c /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/src/mbgl/geometry/line_atlas.cpp
  In file included from /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/src/mbgl/geometry/line_atlas.cpp:2:0:
  /home/bessermt/dev/MyFirstMapboxApp/app/.externalNativeBuild/cmake/debug/mips64/mapbox-gl-prefix/src/mapbox-gl/include/mbgl/gl/gl.hpp:27:23: fatal error: GL/gl.h: No such file or directory
  compilation terminated.
  ninja: build stopped: subcommand failed.
  ninja: build stopped: subcommand failed.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 2 mins 11.039 secs

---


```

**Comment by bessermt** [comment]
Okay, maybe you can just look at my project. I'm just creating the project from:

https://www.mapbox.com/help/first-steps-android-sdk/

You can build it yourself if that helps with the logs you want.

**Comment by mitch-besser** [comment]
Hey @jfirebaugh, did you get a copy of my project?

**Comment by cammace** [comment]
The project you are linking to is a guide to getting started with Mapbox for Android. It uses a stable, prebuilt, version of the master branch. In this case all you'd need to include would be: 

```groovy
repositories {
    mavenCentral()
}

dependencies {
    compile ('com.mapbox.mapboxsdk:mapbox-android-sdk:4.1.1@aar'){
        transitive=true
    }
}
```

It sounds like you might be trying to fork this repo and build some custom work into the library. If this is the case your best option would be to clone this repo and follow the instructions laid out in [CONTRIBUTING_MACOS](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/CONTRIBUTING_MACOS.md) or [CONTRIBUTING_LINUX](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/CONTRIBUTING_LINUX.md). From here you can build your application using the Testapp. module of the project. It also sounds like you are using a custom build script which we do not support. 

For future support questions please reach out using either our [contact page](mapbox.com/contact) or on Stack Overflow.

**Comment by mitch-besser** [comment]
Hi @cammace,

I'm not, you have the wrong link. I've been blocked from editing. 

Let me try again. It is:

https://github.com/bessermt/MyFirstMapboxApp
